### PR TITLE
add fancier success icons

### PR DESCRIPTION
**Pull request by ilrudie**
**Please provide a description of this PR:**

added fancier icons for success statuses in istioctl

### CODE REVIEW

File: operator/pkg/name/name.go [hunk 1]
```diff
@@ -86,6 +86,15 @@ const (
 	IstioOperatorCustomResourceName ComponentName = "IstioOperatorCustomResource"
 )
 
+var IstioComponentSuccessIcons = map[ComponentName]string{
+	IstioBaseComponentName: "⛵️",
+	PilotComponentName:     "🧠",
+	CNIComponentName:       "🚦",
+	ZtunnelComponentName:   "🔒",
+	IngressComponentName:   "🚪",
+	EgressComponentName:    "✋",
```
Comment (linsun) [inline operator/pkg/name/name.go:95]: Maybe use similar or same icons for ingress and egress?

File: operator/pkg/name/name.go [hunk 2]
```diff
@@ -86,6 +86,15 @@ const (
 	IstioOperatorCustomResourceName ComponentName = "IstioOperatorCustomResource"
 )
 
+var IstioComponentSuccessIcons = map[ComponentName]string{
+	IstioBaseComponentName: "⛵️",
+	PilotComponentName:     "🧠",
+	CNIComponentName:       "🚦",
```
Comment (linsun) [inline operator/pkg/name/name.go:92]: Do you think it makes sense to use the [CNI icon](https://www.cni.dev/)?

**Review by linsun — COMMENTED** [review]
I like the istio base, pilot and ztunnel ones, nice!!!

File: operator/pkg/name/name.go (hunk 2 above)
Comment (ilrudie) [reply operator/pkg/name/name.go:92]: Ideally that'd be great but I don't _think_ there is a unicode character for it. I'll take another look though.

File: operator/pkg/name/name.go (hunk 1 above)
Comment (ilrudie) [reply operator/pkg/name/name.go:95]: I thought about 🛫 (plane takeoff/leave) for egress and 🛬 (plane land/arrive) for ingress.

Comment (ilrudie) [reply operator/pkg/name/name.go:95]: @linsun, how do you feel about using the planes?

File: operator/pkg/name/name.go (hunk 2 above)
Comment (ilrudie) [reply operator/pkg/name/name.go]: I tried 🪢 which sort of looks like ethernet cables being woven

File: operator/pkg/name/name.go [hunk 3]
```diff
@@ -86,6 +86,15 @@ const (
 	IstioOperatorCustomResourceName ComponentName = "IstioOperatorCustomResource"
 )
 
+var IstioComponentSuccessIcons = map[ComponentName]string{
+	IstioBaseComponentName: "⛵️",
+	PilotComponentName:     "🧠",
+	CNIComponentName:       "🪢",
+	ZtunnelComponentName:   "🔒",
+	IngressComponentName:   "🛬",
+	EgressComponentName:    "🛫",
```
Comment (linsun) [inline operator/pkg/name/name.go:95]: Hi @ilrudie Is ingress/egress airplane flying icon?

**Review by linsun — COMMENTED** [review]
LGTM

Sending to ambient-dev to see if others have comments.

File: operator/pkg/name/name.go (hunk 3 above)
Comment (ilrudie) [reply operator/pkg/name/name.go:95]: ingress is plane landing, egress is plane leaving

Comment (ilrudie) [reply operator/pkg/name/name.go:95]: not meaning these are perfect but they at least sort of make sense (to me)

**Review by MorrisLaw — APPROVED** [review]
This lgtm, thanks Ian!

**Review by linsun — APPROVED** [review]

**Comment by stevenctl** [comment]
Most of these characters don't render for me. I think seeing squares wouldn't be the best look. Not everyones terminal/font will have full unicode/emoji support. 

⛵️🧠🪢🔒🛬🛫✅❌✅❌

Only 3 of these (the sailboat, the knot and the lock) render for me.

**Comment by hanxiaop** [comment]
In my case, the first letter of the last message is cut off:
image

**Comment by linsun** [comment]
Good it is in master :) thanks @stevenctl and @hanxiaop for reporting it. cc @ilrudie

**Comment by ilrudie** [comment]
We can revert it if folks want and I can look at some detection methods so we can have like a fancy set and a safe set to choose from based on detecting what is/isn't supported

**Comment by ilrudie** [comment]
An alternative might be to use a simple up/down to indicate success as we were previously but decorate the end of the line with fancy characters. this way if we do get the mystery char box at least what occurred is clear and still readable.

**Pull request merged**